chain/bitcoind.go

```diff
@@ -511,40 +512,8 @@ func (c *BitcoindClient) socketHandler() {
 	defer c.wg.Done()
 
 	// Connect a ZMQ socket for block notifications
-	zmqCtx, err := zmq4.NewContext()
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	defer func() {
-		if zmqCtx != nil {
-			err = zmqCtx.Term()
-			log.Infof("ZMQ context terminated: %v\n", err)
-		}
-	}()
-	zmqClient, err := zmqCtx.NewSocket(zmq4.SUB)
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	defer func() {
-		if zmqClient != nil {
-			err = zmqClient.Close()
-			log.Infof("ZMQ socket closed: %v\n", err)
-		}
-	}()
-	err = zmqClient.SetSubscribe("rawblock")
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	err = zmqClient.SetSubscribe("rawtx")
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	err = zmqClient.SetRcvtimeo(c.zmqPollInterval)
-	err = zmqClient.Connect(c.zmqConnect)
+	zmqClient, err := gozmq.Subscribe(c.zmqConnect, []string{"rawblock",
+		"rawtx"}, c.zmqPollInterval)
```

Comment:
We don't to worry about closing the socket or anything like that?

Reply:
Good point, forgot to add that. Thanks!

Reply:
Fixed.